### DISCUSSION TITLE
Newman stop after first test assertion fail

### DISCUSSION THREAD

**Discussion by MarcoFierimonte** [apideck-libraries/portman#289] [Q&A] [answered]
Hi all, thanks for this tool it is really useful !

I have a problem with the newman run, basically I run newman with portman and after the first assertion fail the execution is stopped. 
I remember that newman continue to execute all the api also if there are fails.

**This is the command I run:**
`npx @apideck/portman --cliOptionsFile ./portman-cli-options.json --runNewman --newmanOptionsFile ./newman-options.json
`

**This is the output of the command:**

------------------------------------------------------------------------------------------------------------------------------------------
 1.  AssertionError                      [GET]::/v1/scenario/projects/:projectCode/scenario/:scenarioCode/revisions/:revisionCode/file - Status code is 2xx
                                         expected response code to be 2XX but found 500
                                         at assertion:0 in test-script
                                         inside "/v1/scenario / Download windcoe file"
                                                                                                                                                                                   
 2.  AssertionFailure                    [GET]::/v1/scenario/projects/:projectCode/scenario/:scenarioCode/revisions/:revisionCode/file - Status code is 2xx
                                         at test-script:38:17
                                         inside "/v1/scenario / Download windcoe file"
Newman run failed with: 
[GET]::/v1/scenario/projects/:projectCode/scenario/:scenarioCode/revisions/:revisionCode/file - Status code is 2xx
------------------------------------------------------------------------------------------------------------------------------------------

So it stops after the first fails also if there are many other run to execute. Any idea about how to solve it? 

Thanks.
Marco

**Comment by MarcoFierimonte** [comment]
Hi @thim81 

thanks for your reply.
I am using these newman options (I tried also with defaults values):
```
{
  "iteration-count": 1,
  "suppress-exit-code" : true,
  "ignore-redirects": true,
  "insecure": true
}
```
and the newman embedded in portman. 
I just tried also to run generated collection directly from newman and it worked correctly. 
Let me know if something else is needed.

Thanks.
Marco F.

**Comment by nicklloyd** [comment, marked as answer]
Hi @MarcoFierimonte - by default, Portman passes `abortOnFailure:true` to Newman. You should be able to overwrite this to  see all fails in a run.

**Reply by MarcoFierimonte** [reply]
It worked...thanks @nicklloyd :)

**Reply by ikenna** [reply]
Just for the benefit of others, I did what nicklloyd said by adding the below to my portman-cli-options.json file:
>   "newmanRunOptions": {
>     "abortOnFailure": false
>   }

**Reply by chyde80** [reply]
does not work for me! Any other things I may be missing?

**Reply by thim81** [reply]
The options you are using might not be matching with the Newman options.
Did you have a look at the example available: https://github.com/apideck-libraries/portman/tree/main/examples/cli-options#portman-newman-cli-options.

It could be that Newman changed their options: https://learning.postman.com/docs/running-collections/using-newman-cli/newman-options/
 `bail` seems to be the new option name.

**Reply by chyde80** [reply]
yeah it is "bail" but it's not working. Here is my newman-options file example i'm passing in at command line with newmanOptionsFile
``{
    "environment": "./resources/customermeta local-client.postman_environment.json",
    "ignore-redirects": true,
    "bail": false
}``

**Reply by chyde80** [reply]
Don't you need to fix this, if it changed to `bail`?  
![image](https://user-images.githubusercontent.com/94012770/215874586-e1a74cb0-8e9b-4564-aeab-44c63a018e55.png)

**Reply by chyde80** [reply]
I must have had something incorrect. `bail` does nothing, but having my newman options file like this it works as expected and DOES continue on test failure. Not sure if this is how it SHOULD be - should you rename to `bail`??

![image](https://user-images.githubusercontent.com/94012770/215875533-58359ee2-7c18-453e-b2d7-1ebe2c4351ac.png)

**Reply by nicklloyd** [reply]
Hi @chyde80 - `abortOnFailure` - it looks that newman ties it both to [script errors as well as failing tests](https://github.com/postmanlabs/newman#newmanruncallbackerror-object--summary-object). 

Your first didn't work as `abortOnFailure: true` is the [default](https://github.com/apideck-libraries/portman/blob/main/src/application/newman/runNewmanWith.ts#L36) 

So your second example is working as it is overriding and thereby disabling that option. 

It's my understanding that `bail` is set to false by default, so I think in your case you can omit. 

If you do find we're implementing Newman incorrectly, please submit a PR and we'll review.

**Reply by chyde80** [reply]
I am running in a Jenkins pipeline and am seeing "collection run completed" when the run actually has failed tests. Why is the exit code not coming back correctly w/ an error? I have "abortOnFailure" set to false so that I can run the entire collection w/ tests before failing the pipeline stage. I can see in the console that newman/portman is reporting "Collection run completed" instead of what I would expect from [Newman's docs](https://github.com/postmanlabs/newman#newmanrunevents) which should throw "collection run encountered an error."

**Reply by thim81** [reply]
Which version of Newman are you using? So we can try to reproduce the issue. 
We wrap the Newman CLI in Portman, so perhaps the variable has changed in a newer version of Newman.

**Reply by chyde80** [reply]
looks like the one packaged with portman in node-modules that it pulled in is newman 5.3.2

**Reply by chyde80** [reply]
I opened this issue as well https://github.com/apideck-libraries/portman/issues/504

**Reply by chyde80** [reply]
I made a comment on https://github.com/apideck-libraries/portman/issues/504 to explain how I got around this, but would be good if it could be fixed.

**Comment by thim81** [comment]
hi @MarcoFierimonte,

For your information we improve Portman towards the processing of Newman options in Portman 1.18.1 
See the [changelog](https://github.com/apideck-libraries/portman/blob/main/CHANGELOG.md#v1181---2022-09-30)
